openshift_metrics/merge.py

```diff
@@ -53,6 +55,15 @@ def main():
         nargs="*",
         help="List of timestamp ranges in UTC to ignore in the format 'YYYY-MM-DDTHH:MM:SS,YYYY-MM-DDTHH:MM:SS'"
     )
+    parser.add_argument(
+        "--use-nerc-rates",
+        action="store_true",
+        help="Use rates from the nerc-rates repo",
+    )
+    parser.add_argument("--rate-cpu-su", type=Decimal)
+    parser.add_argument("--rate-gpu-v100-su", type=Decimal)
+    parser.add_argument("--rate-gpu-a100sxm4-su", type=Decimal)
+    parser.add_argument("--rate-gpu-a100-su", type=Decimal)
```

Comment:
The way things are currently structured, it's legal to run with, say, `--use-nerc-rates --rate-cpu-su=1`, even though the second argument would be ignored. This would be a good use case for a [mutually exclusive group](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_mutually_exclusive_group). Alternately, should it be possible to override individual rates from nerc-rates with a command line options?

Since none of the rate options have defaults, they will be `0` if unspecified. Practically, that means that when not using `--use-nerc-rates`, *all* the rate options need to be set explicitly. Should we raise an error if only some of them are set explicitly?

Reply:
@larsks right, I did look into using mutually exclusive group but couldn't get it to work with my use case. I'll take another look at it.

> Since none of the rate options have defaults, they will be 0 if unspecified.

I tested this behavior and it's actually NoneType if unspecified and then when I cast it to Decimal it'll throw an error.

```
  File "/Users/naved/work/openshift-usage-scripts/openshift_metrics/merge.py", line 116, in main
    gpu_a100=Decimal(args.rate_gpu_a100_su),
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: conversion from NoneType to Decimal is not supported
```

So it raises an ugly error if not all of the rates are set but it doesn't silently set it `0`. And now that I think about it, I actually don't need to manually cast it as Decimal since the type is already specified. So, if one of the rates is not specified the program will fail at a later stage when it tries to use this NoneType rate (which would be ugly).

I did think of these scenarios when I was creating this PR and I had a function to manually validate all these conditions but I felt it's getting too complicated for a tool that will mostly be run automatically. But I'd be happy to add the validation back. I am going to first see if I can use the mutually exclusive groups, if not I'll had some custom argument validation for these.

Reply:
> I felt it's getting too complicated for a tool that will mostly be run automatically

Yeah, there's a reason I marked the PR as "approved" :).

> I did look into using mutually exclusive group but couldn't get it to work with my use case

On second thought, it might be tricky to make this behave the way I was thinking.

Reply:
> Yeah, there's a reason I marked the PR as "approved" :).
 
In that case, I am going to go ahead and merge this.